src/modules/urlmarkers/urlmarker-service.js

```diff
@@ -17,177 +18,227 @@ angular.module('anol.urlmarkers')
+         * @ngdoc method
+         * @name setMarkerStyle
+         * @methodOf anol.urlmarkers.UrlMarkersServiceProvider
+         * @param {object} style marker style
+         */
+        this.setMarkerStyle = function (style) {
             _style = style;
         };
 
         /**
-     * @ngdoc method
-     * @name setPopup
-     * @methodOf anol.urlmarkers.UrlMarkersServiceProvider
-     * @param {boolean} usePopup
-     * @description When not using popup a label text is added. This can be styled by markerStyle
-     */
-        this.setUsePopup = function(usePopup) {
+         * @ngdoc method
+         * @name setPopup
+         * @methodOf anol.urlmarkers.UrlMarkersServiceProvider
+         * @param {boolean} usePopup
+         * @description When not using popup a label text is added. This can be styled by markerStyle
+         */
+        this.setUsePopup = function (usePopup) {
             _usePopup = angular.isUndefined(usePopup) ? _usePopup : usePopup;
         };
 
         /**
-     * @ngdoc method
-     * @name setPopupOffset
-     * @methodOf anol.urlmarkers.UrlMarkersServiceProvider
-     * @param {Array.<number>} popupOffset Offset of placed popup. First value is x- second value is y-offset in px
-     */
-        this.setPopupOffset = function(popupOffset) {
+         * @ngdoc method
+         * @name setPopupOffset
+         * @methodOf anol.urlmarkers.UrlMarkersServiceProvider
+         * @param {Array.<number>} popupOffset Offset of placed popup. First value is x- second value is y-offset in px
+         */
+        this.setPopupOffset = function (popupOffset) {
             _popupOffset = angular.isUndefined(popupOffset) ? _popupOffset : popupOffset;
         };
 
-        this.$get = ['$location', 'MapService', 'LayersService', function($location, MapService, LayersService) {
-        /**
-         * @ngdoc service
-         * @name anol.urlmarkers.UrlMarkersService
-         *
-         * @description
-         * Adds markers specified in url. A valid url marker looks like marker=color:ff0000|label:foobar|coord:8.21,53.15|srs:4326
-         */
-            var UrlMarkers = function(defaultSrs, propertiesDelimiter, keyValueDelimiter, style, usePopup, popupOffset) {
-                var self = this;
-                self.features = [];
-                self.defaultSrs = defaultSrs || MapService.view.getProjection();
-                self.propertiesDelimiter = propertiesDelimiter;
-                self.keyValueDelimiter = keyValueDelimiter;
-                self.style = style;
-                self.usePopup = usePopup;
-                self.popupOffset = popupOffset;
-                self.extractFeaturesFromUrl();
-
-                if(self.features.length === 0) {
-                    return;
+        this.$get = ['$location', 'MapService', 'LayersService', function ($location, MapService, LayersService) {
+            /**
+             * @ngdoc service
+             * @name anol.urlmarkers.UrlMarkersService
```

Comment:
minor: file is named `urlmarker-service` (singular) while service is named `urlmarkers-service`(plural).

Reply:
I fix this in the next pr.